packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts

```diff
@@ -69,19 +69,19 @@ export class OdspSummaryUploadManager {
         referenceSequenceNumber: number,
         tree: api.ISummaryTree,
     ): Promise<IWriteSummaryResponse> {
+        const containsProtocolTree = Object.keys(tree.tree).includes(".protocol");
         const { snapshotTree, blobs } = await this.convertSummaryToSnapshotTree(
             parentHandle,
             tree,
             ".app",
-            "",
         );
         const snapshot: IOdspSummaryPayload = {
             entries: snapshotTree.entries!,
             message: "app",
             sequenceNumber: referenceSequenceNumber,
             // no ack handle implies this is initial summary after empty file creation.
             // send container payload so server will use it without a summary op
-            type: parentHandle === undefined ? "container" : "channel",
+            type: containsProtocolTree || parentHandle === undefined ? "container" : "channel",
```

Comment:
i'd like a sperate feature gate here. as we do run with summarizeProtocolTree in stress and it works as a channel upload, and i'd like to keep that working